client.cpp

```diff
@@ -423,7 +423,12 @@ void client::handle_event(short evtype)
         new_evtype = EV_READ;
     }
     if (evbuffer_get_length(m_write_buf) > 0) {
-        assert(finished() == false);
+        if (finished()) {
+            if (m_config->requests > 0 )
+                assert(false && "finished to write all requests but write buffer is not empty");
```

Comment:
why is the `&&`? also note that `assert()` doesn't support providing an error message.
consider replacing the whole if statement whith - 
```cpp
assert(m_config->requests > 0)
```

Reply:
@adi-lb https://stackoverflow.com/questions/3692954/add-custom-messages-in-assert